core/store/src/trie/trie_storage.rs

```diff
@@ -616,24 +617,52 @@ impl TrieStorage for TrieCachingStorage {
     }
 }
 
+fn read_node_from_db(
+    store: &Store,
+    shard_uid: ShardUId,
+    hash: &CryptoHash,
+) -> Result<Arc<[u8]>, StorageError> {
+    let key = TrieCachingStorage::get_key_from_shard_uid_and_hash(shard_uid, hash);
+    let val = store
+        .get(DBCol::State, key.as_ref())
+        .map_err(|_| StorageError::StorageInternalError)?
+        .ok_or_else(|| StorageError::StorageInconsistentState("Trie node missing".to_string()))?;
+    Ok(val.into())
+}
+
 impl TrieCachingStorage {
     fn read_from_db(&self, hash: &CryptoHash) -> Result<Arc<[u8]>, StorageError> {
-        let key = Self::get_key_from_shard_uid_and_hash(self.shard_uid, hash);
-        let val = self
-            .store
-            .get(DBCol::State, key.as_ref())
-            .map_err(|_| StorageError::StorageInternalError)?
-            .ok_or_else(|| {
-                StorageError::StorageInconsistentState("Trie node missing".to_string())
-            })?;
-        Ok(val.into())
+        read_node_from_db(&self.store, self.shard_uid, hash)
     }
 
     pub fn prefetch_api(&self) -> &Option<PrefetchApi> {
         &self.prefetch_api
     }
 }
 
+/// Storage for reading State nodes and values from DB.
```

Comment:
This could be a bit more descriptive, right now it just states what the type name already implies. 

For example,

```suggestion
/// Storage for reading State nodes and values directly from DB.
///
/// This `TrieStorage` implementation has no caches, it just goes to DB.
/// It is useful for background tasks that should not affect chunk processing.
```